alexandria/core/management/commands/fix_copied_file_references.py

```diff
@@ -0,0 +1,90 @@
+import botocore
+from django.core.management.base import BaseCommand
+from tqdm import tqdm
+
+from alexandria.core.models import File
+
+
+class Command(BaseCommand):
+    help = "Fixes copied file references by creating a new copy."
+
+    def add_arguments(self, parser):
+        parser.add_argument("--dry", dest="dry", action="store_true", default=False)
+
+    def handle(self, *args, **options):
+        filename_counts = {}
+        for tmp_file in tqdm(
+            File.objects.filter(variant="original").order_by("created_at")
+        ):
```

Comment:
This could be a huge number of files, and in this way, we're loading them all into memory at the same time.

Querysets can be turned into iterators that load the data batch-by-batch from DB, improving memory usage:

```suggestion
        for tmp_file in tqdm(
            File.objects.filter(variant="original").order_by("created_at").iterator()
        ):
```

Reply:
Added iterator